### PR TITLE
feat(bench): add inter-token latency (ITL) metric to bench engine CLI

### PR DESCRIPTION
**Pull request by ningziwen**
**What this PR does / why we need it**:

Adds inter-token latency (ITL) tracking to `lmcache bench engine`. ITL measures the time between consecutive tokens in a streaming response — a key SLO metric for real-time inference that captures the smoothness of token delivery as perceived by the end user.

The existing CLI reports TTFT and decode speed but not per-token latency variance. This PR adds:
- `inter_token_latency` field in `RequestResult` computed from token timestamps
- Mean/P50/P90/P99 ITL in `FinalStats` and JSON output
- ITL section in the terminal summary output
- Documentation update in `docs/source/cli/bench.rst`

The mean ITL is computed from the same filtered population as percentiles (only requests with `inter_token_latency > 0`), ensuring consistency.

**If applicable**:
- [x] this PR contains user facing changes - docs added
- [x] this PR contains unit tests

---

> [!NOTE]
> **Medium Risk**
> Medium risk because it changes benchmark result schemas (CSV/JSON fields) and adds new timing logic in the streaming request path, which could affect reported metrics or downstream parsers.
> 
> **Overview**
> `lmcache bench engine` now tracks and reports **inter-token latency (ITL)** alongside TTFT and decode speed.
> 
> The request sender records per-chunk timestamps and computes average ITL using server-reported `usage.completion_tokens` (handling speculative decoding), aggregates mean/P50/P90/P99 ITL in `StatsCollector`, and exports `inter_token_latency` in `bench_results.csv` plus ITL fields in the JSON summary and final terminal metrics.
> 
> Also adds a small safety guard when closing the async OpenAI client (ignore `RuntimeError` if the event loop is already closed) and updates docs/tests to cover the new metric.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit ad85e48c886962874ca76f150839666e338a7aaa. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: lmcache/cli/commands/bench/engine_bench/request_sender.py [hunk 1]
```diff
@@ -101,48 +110,81 @@ async def send_request(
...
+                        if ct:
+                            num_output_tokens = ct
+
+                    content = _extract_content(chunk, self._completions_mode)
+                    if content:
+                        now = time.time()
+                        if not first_token_time:
+                            first_token_time = now
+                        tokens.append(content)
+                        token_times.append(now)
```
[the remark this answers is not in the document]
Comment (ningziwen) [reply lmcache/cli/commands/bench/engine_bench/request_sender.py]: Intentionally kept separate. The two paths handle fundamentally different stream types:

- **Standard path**: iterates over `openai.AsyncStream` objects with typed attributes (`chunk.usage.prompt_tokens`, `chunk.choices[0].delta.content`). Extraction uses `getattr()` on SDK objects.
- **Raw SSE path**: iterates over `(content, usage_dict)` tuples from `_raw_sse_stream()`, which already normalizes the mixed SSE formats into a common shape. Extraction uses `dict.get()` on plain dicts.

**Comment by ningziwen** [comment]
@ApostaC @deng451e

**Comment by ApostaC** [comment]
Hey @ningziwen , thanks for the contribution! Just wondering if you can help break the PR into smaller pieces to make them much easier for review. 

Also, regarding ITL, can we make sure that every HTTP response chunk contains exactly 1 token? Otherwise the "ITL" is not really accuate.

**Comment by ningziwen** [comment]
@ApostaC Spilt the PRs. Only ITL change now.

Updated the ITL computation to use the server-reported completion_tokens from the usage field as the denominator instead of the number of SSE chunks. So ITL = (last_chunk_time - first_chunk_time) / (completion_tokens - 1). This is accurate regardless of how many tokens each chunk contains, since the total decode time is divided by the actual token count reported by the server.

File: tests/cli/commands/bench/engine_bench/test_request_sender.py [hunk 2]
```diff
@@ -475,3 +475,31 @@ async def test_close_calls_client_close(self, mock_openai_cls) -> None:
...
+        # Standard
+        import time
+
+        token_times = [time.time()]
+        itl = (
+            0.0
+            if len(token_times) < 2
+            else ((token_times[-1] - token_times[0]) / (len(token_times) - 1))
+        )
+        assert itl == 0.0
```
[the remark this answers is not in the document]
Comment (ningziwen) [reply tests/cli/commands/bench/engine_bench/test_request_sender.py:505]: The existing test_request_sender.py tests already cover send_request end-to-end (they mock the OpenAI client and verify RequestResult fields). Our ITL tests supplement those by verifying the math and edge cases (single token = 0, multiple tokens = correct average). Adding another mocked send_request test specifically for ITL would duplicate the existing coverage.

**Comment by KuntaiDu** [comment]
Quick question: When speculative decoding is turned on, >1 tokens might arrive at the same time. How is ITL being calculated in this case?

**Comment by ningziwen** [comment]
@KuntaiDu ITL is computed as (last_chunk_time - first_chunk_time) / (usage.completion_tokens - 1). We use the server-reported token count from usage.completion_tokens as the denominator, not the number of SSE chunks. So when speculative decoding sends >1 tokens in a single chunk, the denominator still reflects the actual token count. Added a clarifying comment and doc note in the latest commit.

**Comment by ningziwen** [comment]
@ApostaC @deng451e @KuntaiDu Could you pls take a look?
